### ISSUE TITLE
No controller input in some games

### ISSUE THREAD

**Issue by Tasosgemah** [libretro/hatari#50]
Tested Altered Beast. No respond to any inputs.

Tested Alien Syndrome. Can reach character select screen. After that it requires joystick inputs so there is no way to get past the screen.

Also, in the joystick menu there are no controllers detected:

![Altered Beast (Europe)-200501-220754](https://user-images.githubusercontent.com/12124465/80835216-febb6400-8bfa-11ea-9b16-12bcbde50261.png)

Standalone version detect my 360 controllers just fine.

**Comment by boydduk** [comment]
Yep, can never get in game joystick to work.  Mouse works find in and out of the hatari menu, but can never start a game because the joystick inputs are ignored in game running from retroarch.

**Comment by Darknior** [comment]
Yes this emulator is really old :(
So bad no one like on C64 or Amiga update it fine ... now these two emulators are perfect !!! Better than original one !

**Comment by bbbradsmith** [comment]
With #68 I was able to play both Altered Beast and Alien Syndrome.

Altered Beast has a weird issue where if you click the mouse the joystick stops responding, which is strange. I suspect that's a Hatari 1.8.0 issue, but I haven't been able to get Altered Beast to run in either 1.8.0 or the latest standalone. As long as you use gamepad only it seems totally fine though. (Can't duplicate the mouse click issue in STeem or SainT.)

Maybe I should put in a core option to disable/block mouse input for this reason though... Hatari doesn't seem to have any user options to actually "unplug" the mouse, it just seems to automatically handle things based on what the game software requests and whether you're using it, but maybe a core option to just never send mouse input to it would be a useful workaround.

**Comment by Tasosgemah** [comment]
> I was able to play both Altered Beast and Alien Syndrome.

How? I never managed to do that. And i didn't try to use the mouse, i mostly play on couch using a controller.

Was there an update that improved the core?

**Comment by bbbradsmith** [comment]
PR #68 is still pending, it hasn't been accepted, so it won't be fixed in the nightly builds, and the CI build doesn't produce a usable artifact that you could test. I could attach a DLL here if you'd like to test?

**Comment by Tasosgemah** [comment]
Sure, i'll be on my computer for an hour or so

**Comment by bbbradsmith** [comment]
[hatari_libretro_68_test.zip](https://github.com/libretro/hatari/files/5469547/hatari_libretro_68_test.zip) test build of 9a7ca63

**Comment by Tasosgemah** [comment]
Nice, both games now work with the controller. I also noticed there are some extra input options where you can map various keyboard keys on the controller, like the "virtual keyboard space" option. This can skip most intros without bringing the actual virtual keyboard. Not sure if that was a thing before but it's extremely useful.

**Comment by Secamfr** [comment]
Hello @bbbradsmith , i compiled your #68 for raspberry pi and it works very well ! Nice job, i think you can finalize it with space on R3 now, thanks.

**Comment by bbbradsmith** [comment]
Should be fixed with #68

**Comment by borrelnoten** [comment]
Hi, can the new version also be compiled for xbox? I've treid to do this using the current version but I'm unable to compile it using the given instructions (make file using msys 1.0, using powerpckit stuff... not a guru using make at all so gut stuck at some point where the compiler did not find memory.h :( )
